### PR TITLE
Fix dafgda and dafus

### LINKED ISSUE #39: cspyce.dafus does not use array shape inputs nd, ni
`cspyce.dafus` is supposed to return two arrays (one float, one int) of results of a chosen size, as specified by the second and third arguments `nd` and `ni`. The current results always return two arrays of size 256, regardless of the input. Here is a simplified unit test:

```
import cspyce as cs

spk = 'cspyce-unit-test-kernels/de405s_littleendian.bsp'
handle = cs.dafopr(spk)
cs.dafbfs(handle)
found = cs.daffna()
out = cs.dafgs()
dc, ic = cs.dafus(out, 2, 6)
assert len(dc) == 2
```

### PR DESCRIPTION
**Pull request by fyellin**
Fixes #39 

Fix dafgda and dafus by passing them a large array, but letting them have the option of letting us know the actual resulting size.

### CODE REVIEW

**Comment by fyellin** [comment]
Note:  I could also have fixed this by creating definitions for dafus and dafgda in cspyce1.py, that just create the arrays of the size, and pass those to the function. But the TYPEMAP_INOUT's in cspyce_typemaps.i aren't currently used and haven't been tested.

**Comment by rfrenchseti** [comment]
Can you go ahead and fix `dafgs`? Test case is in the associated issue.

**Comment by fyellin** [comment]
Don't approve just yet.  I want to try one more possibility.  I'm going for simple and clean.

**Comment by fyellin** [comment]
Not sure the right way to handle this.  I'm getting the coverage tests failing because dafgda and dafus now have a few lines of handwritten Python, when they didn't before.  If you want to accept this anyway, fine.  If you want to tell me what a test for dafgda and dafus would look like and where they go, I'll add them.

I'm not going to be making any more changes.  I'm happy with where this ended up.  The calculations are now done in Python rather than C, which is where they belonged all along.

**Comment by rfrenchseti** [comment]
Code coverage tests aren't important. We have another branch that will be adding tests for all DAF functions.

File: cspyce/cspyce1.py [hunk 1]
```diff
@@ -810,6 +810,27 @@ def dasec(handle):
 del CSPYCE_RETNAMES['dasec'][1]
 del CSPYCE_DEFINITIONS['dasec']['done']
 
+################################################################################
+# These functions need to be passed a fixed-size array that's based on the value of
+# other arguments.  It is simpler to create those arrays in Python and pass them.
+################################################################################
+
+def dafgda(handle, begin, end):
+    size = abs(end - begin) + 1
```
Comment (rfrenchseti) [inline cspyce/cspyce1.py:819]: Why the `abs`? The function documentation says:

```
   1)  If `baddr' is zero or negative, the error SPICE(DAFNEGADDR)
       is signaled by a routine in the call tree of this routine.
   2)  If baddr > eaddr, the error SPICE(DAFBEGGTEND) is signaled by a
       routine in the call tree of this routine.
```

Wouldn't it be better just to use `size = 0 if begin <= 0 or begin > end else end - begin + 1` ?

File: swig/cspyce0.i [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 #define NPLANE 4
 #define NELLIPSE 9
-#define DAFSIZE 256     // DAF size
+#define DAFSIZE 125      // size of DAF summary
```
Comment (rfrenchseti) [inline swig/cspyce0.i:14]: Comment doesn't line up with next line

File: swig/cspyce_typemaps.i [hunk 3]
```diff
@@ -2565,11 +2563,36 @@ TYPEMAP_ARGOUT(SpiceDouble, NPY_DOUBLE)
...
     (int DIM1, Type INOUT_ARRAY2[][ANY]),
     (int DIM2, Type *INOUT_ARRAY2)
-
 {
     Py_XDECREF(pyarr$argnum);
 }
 
+/*
+ * A LOCAL_INOUT_ARRAY is similar to an INOUT_ARRAY, except that it will have been
+ * generated by * cspyce internally, so we don't need to error check it.
```
Comment (rfrenchseti) [inline swig/cspyce_typemaps.i:2572]: What is "* cspyce"?

**Review by rfrenchseti — CHANGES_REQUESTED** [review]

**Comment by fyellin** [comment]
Fixed.

**Comment by rfrenchseti** [comment]
Fixed but not pushed?

File: cspyce/cspyce1.py (hunk 1 above)
Comment (fyellin) [reply cspyce/cspyce1.py]: I copied code from spiceypy.  I don't know why they did that.  For begin <= 0, I'm happy to just let spice deal with that error.

**Comment by fyellin** [comment]
Apparently gitkraken was confused by something.  It told me it had pushed, but it hadn't.  Sigh.  Things appear better now.  I also merged in the latest changes to master.

**Comment by fyellin** [comment]
If you want, let's put this on hold.  Cleaning up the various das routines made me realize this could be done a little bit more nicely.  I'm just going to make all of those into a single push.  Most of what's here gets re-written again.

**Comment by rfrenchseti** [comment]
Closing until it's ready for further review

**Pull request closed without merging**